rpc.go

```diff
@@ -12,22 +18,99 @@ func (r *RPCError) Error() string {
 
 // RPCRequest represents an incoming JSON-RPC request
 type RPCRequest struct {
-	ID     interface{}    `json:"id" msgpack:"id"`
+	ID     any    `json:"id" msgpack:"id"`
 	Async  bool   `json:"async,omitempty" msgpack:"async,omitempty"`
 	Method string `json:"method,omitempty" msgpack:"method,omitempty"`
-	Params []interface{}  `json:"params,omitempty" msgpack:"params,omitempty"`
+	Params []any  `json:"params,omitempty" msgpack:"params,omitempty"`
 }
 
 // RPCResponse represents an outgoing JSON-RPC response
 type RPCResponse struct {
-	ID     interface{}       `json:"id" msgpack:"id"`
+	ID     any       `json:"id" msgpack:"id"`
 	Error  *RPCError `json:"error,omitempty" msgpack:"error,omitempty"`
-	Result interface{}       `json:"result,omitempty" msgpack:"result,omitempty"`
+	Result any       `json:"result,omitempty" msgpack:"result,omitempty"`
 }
 
 // RPCNotification represents an outgoing JSON-RPC notification
 type RPCNotification struct {
-	ID     interface{}    `json:"id" msgpack:"id"`
+	ID     any    `json:"id" msgpack:"id"`
 	Method string `json:"method,omitempty" msgpack:"method,omitempty"`
-	Params []interface{}  `json:"params,omitempty" msgpack:"params,omitempty"`
+	Params []any  `json:"params,omitempty" msgpack:"params,omitempty"`
+}
+
+type RPCRawResponse struct {
+	id      string
+	Data    []byte
+	Decoded *RPCResponse
+
+	decodedError bool
+	hasError     bool
+	error        *RPCError
```

Comment:
those 3 members look strange to me, hasError should be `error != nil` IMO
and document what decodedError is

Reply:
These are mainly just states so we know that we've processed looking for the error in the raw bytes, 
decodedError = did we try to decode it?
hasError = we decoded it & we found an error